haskell/private/pkgdb_to_bzl.py

```diff
@@ -0,0 +1,170 @@
+        include_dirs = [],
+        library_dirs = [],
+        dynamic_library_dirs = [],
+        depends = [],
+        ld_options = [],
+        extra_libraries = [],
+    )
+    for field in fields:
+        key, value = field.split(":", 1)
+        value = value.strip()
+        if key == "name":
+            pkg.name = value
+        elif key == "version":
+            pkg.version = value
+        elif key == "id":
+            pkg.id = value
+        elif key == "include-dirs":
+            pkg.include_dirs = value.split()
+        elif key == "library-dirs":
+            pkg.library_dirs = value.split()
+        elif key == "dynamic-library-dirs":
+            pkg.dynamic_library_dirs = value.split()
+        elif key == "hs-libraries":
+            pkg.hs_libraries = value.split()
+        elif key == "depends":
+            pkg.depends = value.split()
+        elif key == "ld-options":
+            pkg.ld_options = [opt.strip('"') for opt in value.split()]
+        elif key == "extra-libraries":
+            pkg.extra_libraries = value.split()
+
+    # pkgroot is not part of .conf files. It's a computed value. It is
+    # defined to be the directory enclosing the package database
+    # directory.
+    pkg.pkgroot = os.path.dirname(os.path.dirname(os.path.realpath(conf)))
+
+    pkg_id_map.append((pkg.name, pkg.id))
+    for hs_library in pkg.hs_libraries:
+        output += [
+            textwrap.dedent("""\
+                haskell_import(
+                    name = "{name}",
+                    id = "{id}",
+                    deps = {deps},
+                    hdrs = {hdrs},
+                    includes = {includes},
+                    linkopts = {linkopts},
+                    shared_library = {shared_library}[0] if {shared_library} else None,
+                    static_library = {static_library}[0] if {static_library} else None,
+                    static_profiling_library = {static_profiling_library}[0] if {static_profiling_library} else None,
+                    version = "{version}",
+                    visibility = ["//visibility:public"],
+                )
+                """.format(
+                    name = pkg.name,
+                    id = pkg.id,
+                    version = pkg.version,
+                    hdrs = "glob({})".format([
+                        path_to_label("{}/**/*.h".format(include_dir), pkg.pkgroot)
+                        for include_dir in pkg.include_dirs
+                        if path_to_label(include_dir, pkg.pkgroot)
+                    ]),
+                    includes = [
+                        "/".join([repo_dir, path_to_label(include_dir, pkg.pkgroot)])
+                        for include_dir in pkg.include_dirs
+                        if path_to_label(include_dir, pkg.pkgroot)
+                    ],
+                    static_library = "glob({})".format([
+                        path_to_label("{}/lib{}.a".format(library_dir, hs_library), pkg.pkgroot)
```

Comment:
Doesn't this also catch profiling libraries `libHSxyz_p.a`?

Reply:
In theory no. There is no wildcard in the glob here. The only reason we use `glob` is so that if the library file is missing (some packages don't have *both* a shared and a static library), then it won't error out.

Reply:
Oh, I see. Sorry, assumed there was a wild-card.